javascript/libs/ui/src/lib/modal/components/modal/modal.component.ts

```diff
@@ -81,6 +85,21 @@ export class ModalComponent implements AfterViewInit, OnDestroy {
   @ViewChild('domPortalContent')
   public domPortalContent!: PolymorpheusContent<HTMLElement>;
 
+  @HostListener('document:click', ['$event'])
+  public onDocumentClick(event: Event) {
+    if (this.closeClickOutside) {
+      const target = event.target as HTMLElement;
+      const overlay =
+        document.querySelector('tui-dialog-host')?.children[0] === target;
+      const modalInner = target
```

Comment:
you can access to a data attribute with: `target.dataset.ref`

Reply:
changed